File: app/src/main/java/com/example/kampustangerang/CampusAdapter.java
package com.example.kampustangerang;

import android.view.LayoutInflater;
import android.view.ViewGroup;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.bumptech.glide.Glide;
import com.example.kampustangerang.databinding.ItemCampusBinding;
import java.util.List;

public class CampusAdapter extends RecyclerView.Adapter<CampusAdapter.CampusViewHolder> {

    private List<Campus> campusList;

    public CampusAdapter(List<Campus> campusList) {
        this.campusList = campusList;
    }

    @NonNull
    @Override
    public CampusViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        ItemCampusBinding binding = ItemCampusBinding.inflate(LayoutInflater.from(parent.getContext()), parent, false);
        return new CampusViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull CampusViewHolder holder, int position) {
        Campus campus = campusList.get(position);
        holder.binding.tvCampusName.setText(campus.getName());
        holder.binding.tvCampusLocation.setText(campus.getLocation());
        holder.binding.tvCampusDescription.setText(campus.getDescription());

        Glide.with(holder.itemView.getContext())
                .load(campus.getImageUrl())
                .placeholder(android.R.drawable.ic_menu_gallery)
                .error(android.R.drawable.ic_menu_report_image)
                .into(holder.binding.ivCampus);
    }

    @Override
    public int getItemCount() {
        return campusList.size();
    }

    public static class CampusViewHolder extends RecyclerView.ViewHolder {
        ItemCampusBinding binding;

        public CampusViewHolder(ItemCampusBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }
    }
}

File: app/src/main/java/com/example/kampustangerang/CampusAdapter.kt
package com.example.kampustangerang

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.example.kampustangerang.databinding.ItemCampusBinding

class CampusAdapter(private val campusList: List<Campus>) :
    RecyclerView.Adapter<CampusAdapter.CampusViewHolder>() {

    class CampusViewHolder(val binding: ItemCampusBinding) : RecyclerView.ViewHolder(binding.root)

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CampusViewHolder {
        val binding = ItemCampusBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return CampusViewHolder(binding)
    }

    override fun onBindViewHolder(holder: CampusViewHolder, position: Int) {
        val campus = campusList[position]
        holder.binding.apply {
            tvCampusName.text = campus.name
            tvCampusLocation.text = campus.location
            tvCampusDescription.text = campus.description
            tvFacilities.text = campus.facilities
            tvRegistrationDate.text = campus.registrationDate

            Glide.with(holder.itemView.context)
                .load(campus.imageUrl)
                .placeholder(android.R.drawable.ic_menu_gallery)
                .error(android.R.drawable.ic_menu_report_image)
                .into(ivCampus)
        }
    }

    override fun getItemCount(): Int = campusList.size
}
